static/js/admin/blocklist_blocklistsubmission.js

```diff
@@ -17,33 +17,60 @@ document.addEventListener('DOMContentLoaded', () => {
       a.addEventListener('click', checkOrClearAllCheckboxes, true),
     );
 
-  const enableIfChecked = (isChecked, fieldId) => {
-    const field = document.querySelector(`#${fieldId}`);
-    if (isChecked) {
-      field.removeAttribute('disabled');
-    } else {
-      field.setAttribute('disabled', true);
-    }
+  const enableIfChecked = (isChecked, fieldsSelector) => {
+    document.querySelectorAll(fieldsSelector).forEach((field) => {
+      if (isChecked) {
+        field.removeAttribute('disabled');
+      } else {
+        field.setAttribute('disabled', true);
+      }
+    });
   };
 
-  document
-    .querySelector('#id_update_reason_value')
-    .addEventListener('click', (e) =>
-      enableIfChecked(e.target.checked, 'id_reason'),
+  const updateReasonValueCheckbox = document.getElementById(
+    'id_update_reason_value',
+  );
+  if (updateReasonValueCheckbox) {
+    updateReasonValueCheckbox.addEventListener('click', (e) =>
+      enableIfChecked(e.target.checked, '#id_reason'),
     );
+    updateReasonValueCheckbox.addEventListener('click', (e) =>
+      enableIfChecked(e.target.checked, 'input[name="canned_reasons"]'),
+    );
+    enableIfChecked(updateReasonValueCheckbox.checked, '#id_reason');
+    enableIfChecked(
+      updateReasonValueCheckbox.checked,
+      'input[name="canned_reasons"]',
+    );
+  }
 
-  document
-    .querySelector('#id_update_url_value')
-    .addEventListener('click', (e) =>
-      enableIfChecked(e.target.checked, 'id_url'),
+  const updateUrlValueCheckbox = document.getElementById('id_update_url_value');
+  if (updateUrlValueCheckbox) {
+    updateUrlValueCheckbox.addEventListener('click', (e) =>
+      enableIfChecked(e.target.checked, '#id_url'),
     );
+    enableIfChecked(updateUrlValueCheckbox.checked, '#id_url');
+  }
 
-  enableIfChecked(
-    document.querySelector('#id_update_reason_value').checked,
-    'id_reason',
-  );
-  enableIfChecked(
-    document.querySelector('#id_update_url_value').checked,
-    'id_url',
-  );
+  const fillCannedReason = (e) => {
+    const reasonField = document.getElementById('id_reason');
+    const cannedReason = e.target.getAttribute('data-block-reason');
+    if (e.target.checked) {
+      reasonField.setRangeText(
+        cannedReason,
+        reasonField.selectionStart,
+        reasonField.selectionEnd,
+        'end',
+      );
+    } else {
+      // Attempt to remove the canned response related to the reason.
+      reasonField.value = reasonField.value.replace(cannedReason, '');
+    }
+  };
+
+  document
+    .querySelectorAll('#id_canned_reasons input')
+    .forEach((a) => a.addEventListener('click', fillCannedReason));
+  // Move the cursor to the end of the textarea
+  document.getElementById('id_reason').setSelectionRange(100000, 100000);
```

Comment:
eh well, that's one way to do it...

Reply:
heh, I guess I could have got the length of the field value but this worked too